File: CPP/Kode Program/NasiGoreng.cpp
#ifndef NASIGORENG_CPP
#define NASIGORENG_CPP

#include <iostream>
#include <string>
#include "Items.cpp"

using namespace std;

class NasiGoreng : public Items
{
    private:
        string Jenis;            // Contoh: "Ayam", "Seafood"
        int Lvl_Kepedasan;        // Level kepedasan dari 1 - 5

    public:
        // Constructor Default
        NasiGoreng() 
        {

        }

        // Setter dan Getter Jenis
        void setJenis(string jenis1)
        {
            Jenis = jenis1;
        }

        string getJenis()
        {
            return Jenis;
        }

        // Setter dan Getter Lvl_Kepedasan
        void setLvlKepedasan(int lvl)
        {
            if (lvl >= 1 && lvl <= 5) {  // Validasi agar level kepedasan berada dalam rentang 1 - 5
                Lvl_Kepedasan = lvl;
            } else {
                cout << "Level kepedasan tidak valid! (Pilih antara 1 - 5)" << endl;
            }
        }

        int getLvlKepedasan()
        {
            return Lvl_Kepedasan;
        }

        ~NasiGoreng()
        {

        }
};

#endif  // NASIGORENG_CPP

File: Python/Kode Program/NasiGoreng.py
from Items import Items

class NasiGoreng(Items):
    def __init__(self, Id="", Nama="", Kategori="", HargaProduk=0, Jenis="", Lvl_Kepedasan=0):
        super().__init__(Id, Nama, Kategori, HargaProduk)
        self.__Jenis = Jenis
        self.__Lvl_Kepedasan = Lvl_Kepedasan

    def set_jenis(self, jenis):
        self.__Jenis = jenis

    def get_jenis(self):
        return self.__Jenis

    def set_lvl_kepedasan(self, lvl):
        self.__Lvl_Kepedasan = lvl

    def get_lvl_kepedasan(self):
        return self.__Lvl_Kepedasan
